common/scala/src/main/scala/whisk/http/PoolingRestClient.scala

```diff
@@ -55,16 +63,18 @@ class PoolingRestClient(protocol: String, host: String, port: Int, queueSize: In
     Http().cachedHostConnectionPoolHttps[Promise[HttpResponse]](host = host, port = port)
   }
 
+  private val defaultHttpFlow = pool.mapMaterializedValue { x =>
+    poolPromise.success(x); x
+  }
+
   private val poolPromise = Promise[HostConnectionPool]
 
   // Additional queue in case all connections are busy. Should hardly ever be
   // filled in practice but can be useful, e.g., in tests starting many
   // asynchronous requests in a very short period of time.
   private val requestQueue = Source
     .queue(queueSize, OverflowStrategy.dropNew)
-    .via(pool.mapMaterializedValue { x =>
-      poolPromise.success(x); x
-    })
+    .via(httpFlow.getOrElse(defaultHttpFlow))
     .toMat(Sink.foreach({
       case ((Success(response), p)) => p.success(response)
```

Comment:
I can’t  tell just from looking at these few lines of the one promise is being competed twice or if it’s two different promises.

Reply:
@rabbah, do you mean `poolPromise.success(x); x`?